libdrgn/arch_aarch64.c

```diff
@@ -408,7 +409,7 @@ linux_kernel_pgtable_iterator_next_aarch64(struct drgn_program *prog,
 	int level;
 	uint16_t num_entries = it->last_level_num_entries;
 	uint64_t table = it->it.pgtable;
-	bool table_physical = false;
+	bool table_physical = it->it.pgtable_phys;
```

Comment:
I haven't delved deep into the pgtable code, so I was not aware that `table_physical` was here for pretty much all of them and that it would be that easy to properly support a physical page table base.

I had a hack where I fooled drgn into using a physical `swapper_pg_dir` address. But your way is the proper way, I like it a lot!

Reply:
Longer term I was thinking that we might want to try removing the `pgtable_phys` flag and have all page table iterator accesses go via physical memory. That should simplify the code a bit. This seems like it should also work with core dumps because the physical address can be calculated from the program headers in the core dump.